include/swift/AST/DiagnosticsClangImporter.def

```diff
@@ -157,17 +157,22 @@ NOTE(reference_passed_by_value, none, "function uses foreign reference type "
                                       "'import_reference' contract (outlined in "
                                       "C++ Interop User Manual).",
      (StringRef, StringRef))
+
 NOTE(record_not_automatically_importable, none, "record '%0' is not "
                                                 "automatically importable: %1. "
                                                 "Refer to the C++ Interop User "
                                                 "Manual to classify this type.",
      (StringRef, StringRef))
-NOTE(projection_not_imported, none, "C++ method '%0' that returns unsafe "
-                                    "projection of type '%1' not imported",
-     (StringRef, StringRef))
-NOTE(dont_use_iterator_api, none, "C++ method '%0' that returns an unsafe "
-                                  "iterator not imported: use Swift Sequence "
-                                  "APIs instead",
+
+NOTE(projection_not_imported, none, "C++ method '%0' is unavailable in Swift "
+                                    "as it may return or store an address of a memory location "
+                                    "within a temporary copy of the receiver "
+                                    "that is destroyed immediately after the call",
+     (StringRef))
```

Comment:
This seems fairly wordy, what do you think about splitting it up into two notes? One says that the method is unavailable and the other explains why.

Reply:
Having two notes at the same source location is a bit odd. If we could make the error say "C++ method is unavailable in Swift" that would be ideal.

Reply:
That's a really great idea. I think we could implement that. It would be much simpler and more clear for our users.